CefSharp.Core/ManagedCefBrowserAdapter.cpp

```diff
@@ -0,0 +1,32 @@
+#include "Stdafx.h"
+#include "Cef.h"
+
+double ManagedCefBrowserAdapter::GetZoomLevel()
+{
+    auto browser = _clientAdapter->GetCefBrowser();
+
+    if (browser != nullptr)
+    {
+        auto host = browser->GetHost();
+        if (CefCurrentlyOn(TID_UI))
+        {
+            return host->GetZoomLevel();
+        }
+        else
+        {
+            // TODO: Add an async version of GetZoomLevel at some point.
+            // NOTE: Use of ManualResetEvent is required here in order
+            // for simple marshaling of some kind of synchronization primitive
+            // to the callback.
+            Task<double>^ task = Cef::UIThreadTaskFactory->StartNew(gcnew Func<double>(this, &ManagedCefBrowserAdapter::GetZoomLevel));
```

Comment:
As this is hidden away, I'd probably move this into the `C#` world. Thoughts?  The `ChromiumWebBrowser` implementations can deal with as required. I think we'd be better to migrate the users over to using the `async` version. This would also coincide with removing the `ZoomLevel` from `IWebBrowser` and replacing with `GetZoomLevel`. 

Thoughts?